library/general/src/lib/ui/service_status.rb

```diff
@@ -17,46 +17,40 @@
 #  you may find current contact information at www.suse.com
 
 require "yast"
-Yast.import "Service"
 Yast.import "UI"
 
 module UI
-  # Component encapsulating the widgets for managing the status of services (both
-  # currently and on system boot) and the behavior associated to those widgets
+  # Widgets for managing the status of services (both currently and on system
+  # boot) and the behavior associated to those widgets
   #
-  # TODO: The API is experimental and subject to change. This component is only
-  # used by yast2-dns-server and should not be used by any other module until
-  # an upcoming refactoring, planned for SLE12-SP2. To see some of the planned
-  # improvements, check https://github.com/yast/yast-yast2/pull/401
-  #
-  # As long as #handle_input is invoked in the event loop, the component will
-  # handle interactive starting and stopping of the service on user demand. In
-  # addition #reload can be used after saving the settings.
-  #
-  # To manage the status on boot, the component can be queried for the user
-  # selection using #enabled?. In addition enabled_callback (in constructor)
-  # can be used to observe the status of the corresponding field in the UI.
-  class SrvStatusComponent
+  # As long as #handle_input is invoked in the event loop, the widget will
+  # handle interactive starting and stopping of the service on user demand.
+  class ServiceStatus
     include Yast::UIShortcuts
     include Yast::I18n
     include Yast::Logger
 
-    # @param service_name [String] name of the service as expected by
-    #   Yast::Service
-    # @param reload [Boolean] initial value for the "reload" checkbox.
-    #   Keep in mind it will always be displayed as unchecked if the service is
-    #   not running, despite the real value.
-    # @param enabled_callback [Proc] callback executed when the "enabled on
-    #   boot" checkbox is changed. The only parameter of the callback is the new
-    #   state of the checkbox (boolean).
-    def initialize(service_name, reload: true, enabled_callback: nil)
-      @service_name = service_name
+    # @param service [Object] An object providing the following methods:
+    #   #name, #start, #stop, #enabled?, #running?
+    #   For systemd compliant services, just do
+    #   Yast::SystemdService.find("name_of_the_service")
+    # @param reload [Boolean] Initial value for the "reload" checkbox.
+    #   Keep in mind it will always be displayed as unchecked if the service
+    #   is not running, despite the real value.
+    # @param reload_label [Symbol] Type of label for the "reload" checkbox.
+    #   :restart means that the service will be restarted, not just reloaded.
+    def initialize(service, reload: true, reload_label: :reload)
```

Comment:
@mvidner See screenshot at https://cloud.githubusercontent.com/assets/3638289/11421803/2d793940-9437-11e5-918b-1d31b56307f6.png

There is a checkbox there for the users to specify their want to (reload/restart) after saving changes.
The argument `reload` is the initial value for that checkbox. The `reload_label` argument is used to customize (to some extend, I didn't wanted to allow arbitrary customization) the label of that checkbox.

Doing the real restart/reload is not direct responsibility of the widget, since it's something that will happen after clicking the ok/finish/next/whatever button on the caller dialog.